internal/check/check.go

```diff
@@ -423,3 +425,11 @@ func NotPresent[T any, QT ygnmi.SingletonQuery[T]](query QT) Validator {
 		return nil
 	})
 }
+
+// UnorderedEqual function is used to compare slices in un ordered way.
+func UnorderedEqual[T any, QT ygnmi.SingletonQuery[T]](query QT, want T, sortSliceCompFunc interface{}) Validator {
+	return Predicate(query, fmt.Sprintf("want %#v", want), func(got T) bool {
+		// Sort slices to compare them in unorderd way.
+		return cmp.Equal(got, want, cmpopts.SortSlices(sortSliceCompFunc))
+	})
+}
```

Comment:
Sorry yesterday I was in a hurry to leave the office, so I didn't have time to give a more detailed comment.

Since these functions use generics, let's avoid the any interface by making the type more precise. Here the item is T, so the sorting is between two T's, but the got and want are []T.

```suggestion
// UnorderedEqual function is used to compare a slice of T in an unordered way.
func UnorderedEqual[T any, QT ygnmi.SingletonQuery[[]T]](query QT, want []T, less func(a, b T) bool) Validator {
	return Predicate(query, fmt.Sprintf("want %#v", want), func(got []T) bool {
		// Sort slices to compare them in unorderd way.
		return cmp.Equal(got, want, cmpopts.SortSlices(less))
	})
}
```

Reply:
updated, thank you.